overlord/snapstate/check_snap.go

```diff
@@ -113,6 +113,14 @@ func checkSnap(st *state.State, snapFilePath string, si *snap.SideInfo, curInfo
 	if s.NeedsDevMode() && !flags.DevModeAllowed() {
 		return fmt.Errorf("snap %q requires devmode or confinement override", s.Name())
 	}
+	if s.NeedsClassic() {
+		if !release.OnClassic {
+			return fmt.Errorf("snap %q requires classic confinement which is only available on classic systems", s.Name())
+		}
+		if !flags.Classic {
+			return fmt.Errorf("snap %q requires consent to use classic confinement", s.Name())
```

Comment:
there's a problem with this error message, in that it doesn't really tell you how to fix it. Tricky, because you can't really say `"use --classic blah blah"` because snapweb doesn't have `--classic` :-)
Maybe `snap %q requires classic confinement`? Dunno.